neon-animated-pages.html

```diff
@@ -81,54 +89,64 @@
     listeners: {
       'neon-animation-finish': '_onNeonAnimationFinish'
     },
-
-    _selectedChanged: function(selected) {
-      var selectedPage = this.items[this.selected];
-      var oldPage = this.items[this._prevSelected];
-
-      this._prevSelected = this.selected;
-
-      if (!oldPage) {
+    
+    _selectedChanged: function () {
+      
+      var selectedPage = this.selectedItem;
+      var oldPage = this._prevSelected || false;
+      this._prevSelected = selectedPage;
+      
+      // on initial load and if animateFirstSelected is negated, simply display selectedPage
+      if (!oldPage && !this.animateFirstSelected) {
         this._completeSelectedChanged();
         return;
       }
-
-      // cancel the currently running animation if one is ongoing.
-      if (oldPage.classList.contains('neon-animating')) {
-        this._squelchNextFinishEvent = true;
-        this.cancelAnimation();
-        this._completeSelectedChanged();
-      }
-
-      selectedPage.classList.add('neon-animating');
-      // display the oldPage during the transition.
-      oldPage.classList.add('neon-animating');
-
-      // configure the animation
+      
+      // insert safari fix
       this.animationConfig = [{
         name: 'opaque-animation',
         node: selectedPage
       }];
 
+      // configure selectedPage animations
       if (selectedPage.getAnimationConfig) {
         this.animationConfig.push({
           animatable: selectedPage,
           type: 'entry'
         });
+      }      
+      
+      // configure oldPage animations iff exists
+      if (oldPage) {
+        
+        // cancel the currently running animation if one is ongoing.
+        if (oldPage.classList.contains('neon-animating')) {
+          this._squelchNextFinishEvent = true;
+          this.cancelAnimation();
+          this._completeSelectedChanged();
+        }
+        
+        if (oldPage.getAnimationConfig) {
+          this.animationConfig.push({
+            animatable: oldPage,
+            type: 'exit'
+          });
+        }
+        
+        // display oldPage during transition
+        oldPage.classList.add('neon-animating');
       }
 
-      if (oldPage.getAnimationConfig) {
-        this.animationConfig.push({
-          animatable: oldPage,
-          type: 'exit'
-        });
-      }
-
+      selectedPage.classList.add('neon-animating');
+      
       if (this.animationConfig.length > 1) {
-        this.playAnimation();
+        //this.playAnimation();
+        // in the event of first load, we want to run animations only after current microtasks are complete (ie element attached)
+        this.async(this.playAnimation);
```

Comment:
Can it only async if one of the pages is not attached? Check for `isAttached` maybe?